src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,20 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws InvalidUserException {
+        if (user.getLogin() == null || user.getAge() == null || user.getPassword() == null
+                || user.getLogin().length() < 6 || user.getPassword().length() < 6
+                || user.getAge() < 18 || storageDao.get(user.getLogin()) != null) {
```

Comment:
Consider breaking down the validation logic into separate methods for readability and maintainability. For example, you could have methods like `isLoginValid`, `isPasswordValid`, `isAgeValid`, and `isLoginUnique`.